src/Microsoft.ML.Torch/TorchScoringTransformer.cs

```diff
@@ -0,0 +1,428 @@
+                    if (colShapeLength > 1)
+                    {
+                        if (colShapeLength != _parent.InputShapes[i].Length)
+                            throw Host.Except($"Input shape mismatch: Input Column '{_parent.InputColumnNames[i]}' vector shape length {colShapeLength} does not match expected {_parent.InputShapes[i].Length}.");
+
+                        for (int j = 0; j < colShapeLength; j++)
+                        {
+                            if (colTypeDims[j] != _parent.InputShapes[i][j])
+                                throw Host.Except($"Input shape mismatch: Input Column '{_parent.InputColumnNames[i]}' dimension {j} of size {colTypeDims[j]} does not match expected size {_parent.InputShapes[j]}.");
+                        }
+                    }
+                    else
+                    {
+                        // If the column in the input schema is one dimension we make sure that the total size of the TF shape matches.
+                        long valCount = _parent.InputShapes[i].Aggregate(1, (long x, long y) => x * y);
+
+                        if (vectorType.Size != valCount)
+                            throw Host.Except($"Input size {vectorType.Size} does not match with expect size {valCount}.");
+                        _needReshape[i] = true;
+                    }
+                }
+            }
+
+            protected override DataViewSchema.DetachedColumn[] GetOutputColumnsCore()
+            {
+                return new DataViewSchema.DetachedColumn[]
+                {
+                    // REVIEW: we need to deal with how we get the output size for the vector.
+                    new DataViewSchema.DetachedColumn(_parent.OutputColumnName, new VectorDataViewType(NumberDataViewType.Single), null)
+                };
+            }
+
+            protected override Delegate MakeGetter(DataViewRow input, int iinfo, Func<int, bool> activeOutput, out Action disposer)
+            {
+                Host.AssertValue(input);
+                var inputTensors = new ITorchTensor<float>[_inputColIndices.Length];
+
+                ValueGetter<VBuffer<float>> valuegetter = (ref VBuffer<float> dst) =>
+                {
+                    for (int i = 0; i < _inputColIndices.Length; i++)
+                        inputTensors[i] = CreateTensorValueGetterVec<float>(input, _inputColIndices[i], _parent.InputShapes[i]);
+
+                    ITorchTensor<float> result = _parent.Module.Forward(inputTensors);
+                    var resultSize = result.NumberOfElements;
+                    var editor = VBufferEditor.Create(ref dst, (int)resultSize);
+
+                    result.Data.CopyTo(editor.Values);
+                    dst = editor.Commit();
+                    result.Dispose();
+                };
+
+                // REVIEW: not sure we need the below. Alos, why are we not disposing the input tensors?
+                disposer = () =>
```

Comment:
We actually have to I believe. This is related to the struct vs class discussion we had. I think that calling dispose explicitly is safer.